src/openloco/windows/industry_list.cpp

```diff
@@ -1,16 +1,1323 @@
+            // TODO: only needs to be called once.
+            common::init_events();
+
+            window->current_tab = 0;
+            window->invalidate();
+
+            window->widgets = industry_list::widgets;
+            window->enabled_widgets = industry_list::enabledWidgets;
+
+            window->activated_widgets = 0;
+            window->holdable_widgets = 0;
+
+            window->call_on_resize();
+            window->call_prepare_draw();
+            window->init_scroll_widgets();
+        }
+        return window;
+    }
+
+    namespace new_industries
+    {
+
+        static const gfx::ui_size_t window_size = { 578, 172 };
+
+        enum widx
+        {
+            scrollview = 6,
+        };
+
+        const uint64_t enabledWidgets = common::enabledWidgets | (1 << scrollview);
+
+        widget_t widgets[] = {
+            commonWidgets(577, 171, string_ids::title_fund_new_industries),
+            make_widget({ 3, 45 }, { 549, 111 }, widget_type::scrollview, 1, 2),
+            widget_end(),
+        };
+
+        static window_event_list events;
+
+        // 0x0045819F
+        static void prepare_draw(window* self)
+        {
+            common::prepare_draw(self);
+
+            self->widgets[widx::scrollview].right = self->width - 4;
+            self->widgets[widx::scrollview].bottom = self->height - 14;
+
+            if (is_editor_mode())
+            {
+                self->widgets[common::widx::caption].text = string_ids::title_build_new_industries;
+                self->widgets[common::widx::tab_new_industry].tooltip = string_ids::tooltip_build_new_industries;
+            }
+            else
+            {
+                self->widgets[common::widx::caption].text = string_ids::title_fund_new_industries;
+                self->widgets[common::widx::tab_new_industry].tooltip = string_ids::tooltip_fund_new_industries;
+            }
+        }
+
+        // 0x0045826C
+        static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+
+            if (self->var_83C == 0)
+            {
+                auto xPos = self->x + 3;
+                auto yPos = self->y + self->height - 13;
+                auto width = self->width - 19;
+                gfx::draw_string_494BBF(*dpi, xPos, yPos, width, colour::black, string_ids::no_industry_available);
+                return;
+            }
+
+            auto industryId = self->var_846;
+
+            if (industryId == 0xFFFF)
+            {
+                industryId = self->row_hover;
+
+                if (industryId == 0xFFFF)
+                    return;
+            }
+
+            auto industryObj = objectmgr::get<industry_object>(industryId);
+            auto industryCost = 0;
+
+            if (self->var_846 == 0xFFFF)
+                industryCost = dword_E0C39C;
+
+            if ((self->var_846 == 0xFFFF && dword_E0C39C == (1ULL << 31)) || self->var_846 != 0xFFFF)
```

Comment:
(1ULL << 31) when talking about money this is almost always used as the null value. It is often used from game commands to indicate failure as well.